File: docflow-frontend/types/api/UnauthenticatedAccessAuditCursorPageDTO.ts
import { UnauthenticatedAccessAuditRow } from "./UnauthenticatedAccessAuditRow";

export type UnauthenticatedAccessAuditCursorPageDTO = {
  items: UnauthenticatedAccessAuditRow[];
  hasMore: boolean;
  nextCursorTimestamp: string | null;
  nextCursorId: string | null;
};

File: docflow-backend/src/main/java/com/brutecx/docflow_backend/api/dto/audit/UnauthenticatedAccessAuditCursorPageDTO.java
package com.brutecx.docflow_backend.api.dto.audit;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record UnauthenticatedAccessAuditCursorPageDTO(
        List<UnauthenticatedAccessAuditDTO> items,
        boolean hasMore,
        Instant nextCursorTimestamp,
        UUID nextCursorId
) implements BaseAuditCursorPageDTO<UnauthenticatedAccessAuditDTO> {}
